<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="dobot_magician">


  <!-- Definicje materiałów (DODANE) -->
  <material name="grey">
    <color rgba="0.7 0.7 0.7 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>


  <xacro:property name="robot_params" value="${xacro.load_yaml('$(find my_robot_rviz)/config/robot_params.yaml')}"/>
  <xacro:property name="base_height" value="${robot_params['links']['base_height']}"/>
  <xacro:property name="link1_length" value="${robot_params['links']['link1_length']}"/>
  <xacro:property name="link1_radius" value="${robot_params['links']['link1_radius']}"/>
  <xacro:property name="base_size" value="${robot_params['links']['base_size']}"/>
  <xacro:property name="link2_length" value="${robot_params['links']['link2_length']}"/>
  <xacro:property name="joint_offset1" value="${robot_params['links']['joint_offset1']}"/>
  <xacro:property name="link3_length" value="${robot_params['links']['link3_length']}"/>
  <xacro:property name="joint_offset2" value="${robot_params['links']['joint_offset2']}"/>
  
	<!-- Baza robota -->	
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.2 0.2 ${base_height}"/>
      </geometry>
      <material name="grey" />
    </visual>
  </link>

<!-- Złącze 1: Obrót bazy -->
  <joint name="base_to_link1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.5708" upper="1.5708" effort="100" velocity="1.0"/>
    <origin xyz="0 0 0.069" rpy="0 0 0"/>
  </joint>

  <!-- baza obrotowa  -->
  <link name="link1">
    <visual>
      <geometry>
        <cylinder radius="${link1_radius}" length="${link1_length}"/>
      </geometry>
      <material name="blue" />
    </visual>
  </link>


  <!-- Złącze 2:Ramie -->
    <joint name="link1_to_link2" type="revolute">
      <parent link="link1"/>
      <child link="link2"/>
      <axis xyz="1 0 0"/>
      <limit lower="-1.5708" upper="1.5708" effort="100" velocity="1.0"/>
      <origin xyz="0 0 0.044" rpy="0.78 0 0"/>
    </joint>

  <!-- ramie 1  -->
   <link name="link2">
    <visual>
      <origin xyz="0 0 ${joint_offset1}" rpy="0 0 0"/>
      <geometry>
        <box size="0.03 0.03 ${link2_length}"/>
      </geometry>
      <material name="grey" />
    </visual>
  </link>

  <!-- Złącze 3:Przedramie -->
    <joint name="link2_to_link3" type="revolute">
      <parent link="link2"/>
      <child link="link3"/>
      <axis xyz="1 0 0"/>
      <limit lower="-1.5708" upper="1.5708" effort="100" velocity="1.0"/>
      <origin xyz="0 0 0.135" rpy="0.78 0 0"/>
    </joint>

  <!-- przedramie  -->
  <link name="link3">
   <visual>
   	<origin xyz="0 0 ${joint_offset2}" rpy="0 0 0"/> 
      <geometry>
        <box size="0.03 0.03 ${link3_length}"/>
      </geometry>
      <material name="grey" />
    </visual>
  </link>

  <!-- Złącze 4: uchwyt narzedzia -->
  <joint name="link3_to_link4" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <axis xyz="1 0 0"/>
    <limit lower="-1.5708" upper="1.5708" effort="100" velocity="1.0"/>
    <origin xyz="0 0 ${link3_length}" rpy="0.78 0 0"/>
  </joint>

  <!-- nadgarstek  -->
    <link name="link4">
   <visual>
   	<origin xyz="0 0 0.03" rpy="0 0 0"/> 
      <geometry>
        <box size="0.01 0.01 0.06"/>
      </geometry>
      <material name="grey" />
    </visual>
  </link>

  <!-- Złącze 5: narzedzie -->
  <joint name="link4_to_link5" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <axis xyz="0 0 1"/>
    <limit lower="-1.5708" upper="1.5708" effort="100" velocity="1.0"/>
    <origin xyz="0 0 0.06" rpy="0 0 0"/>
  </joint>

  <!-- dlon  -->
    <link name="link5">
   <visual>
   	<origin xyz="0 0 0.005" rpy="0 0 0"/> 
      <geometry>
        <box size="0.07 0.05 0.01"/>
      </geometry>
      <material name="grey" />
    </visual>
  </link>

</robot>
